async-nats/src/client.rs

```diff
@@ -574,10 +532,10 @@ impl Request {
     /// # #[tokio::main]
     /// # async fn main() -> Result<(), async_nats::Error> {
     /// let client = async_nats::connect("demo.nats.io").await?;
-    /// let request = async_nats::Request::new()
+    /// client.request("service".into(), "data".into())
```

Comment:
Payload isn't really optional because request takes subject and payload. Deprecate this going forward?

Reply:
We sent a lot of requests without payloads.
It's a nice API, but it is true that it's simple to send a request with explicitly empty payload...